x-pack/plugins/apm/public/components/shared/Links/url_helpers.ts

```diff
@@ -60,19 +60,34 @@ const DEFAULT_KIBANA_TIME_RANGE = {
   }
 };
 
-function getQueryWithRisonParams(location: Location, query: RisonDecoded = {}) {
+function getQueryWithRisonParams(
+  location: Location,
+  pathname: string,
+  query: RisonDecoded = {}
+) {
   // Preserve current _g and _a
   const currentQuery = toQuery(location.search);
   const decodedG = risonSafeDecode(currentQuery._g);
   const combinedG = { ...DEFAULT_KIBANA_TIME_RANGE, ...decodedG, ...query._g };
   const encodedG = rison.encode(combinedG);
   const encodedA = query._a ? rison.encode(query._a) : '';
 
-  return {
+  const nextQuery: StringMap = {
     ...query,
-    _g: encodedG,
-    _a: encodedA
+    _g: encodedG
   };
+
+  // Preserve kuery for apm links
+  const isApmLink = pathname.includes('app/apm') || pathname === '';
```

Comment:
When is pathname `=== ''` for an APM link?

Reply:
We often omit it in `KibanaLink`:
https://github.com/elastic/kibana/blob/166c192c7fbe5010dcfa982515a7c4072d34f712/x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/FlyoutTopLevelProperties.tsx#L34-L36

It works either way. But it is maybe a bit weird... Anyway, I had to support it here.

Reply:
Oh, I see -- no pathname means "just stay in APM and go to this new hash". Got it.